contentcuration/contentcuration/utils/recommendations.py

```diff
@@ -61,67 +70,284 @@ def create_backend(self) -> Backend:
+        recommendations into the RecommendationsCache table, ignoring any conflicts.
+
+        :param request: The request to cache.
+        :param response: The response to cache.
+        :return: A boolean indicating whether the caching was successful.
+        :rtype: bool
+        """
+
+        try:
+            nodes = self._extract_data(response)
+            request_hash = self._generate_request_hash(request)
+            existing_cache = set(RecommendationsCache.objects.filter(request_hash=request_hash)
+                                 .values_list('contentnode_id', flat=True))
+            override_threshold = self._extract_override_threshold(request)
+            new_cache = [
+                RecommendationsCache(
+                    request_hash=request_hash,
+                    contentnode_id=node['contentnode_id'],
+                    rank=node['rank'],
+                    override_threshold=override_threshold,
+                ) for node in nodes if node['contentnode_id'] not in existing_cache
+            ]
+            RecommendationsCache.objects.bulk_create(new_cache, ignore_conflicts=True)
+            return True
+        except Exception as e:
+            logging.exception(e)
+            return False
+
+    def _extract_override_threshold(self, request) -> bool:
+        """
+        Extracts the override_threshold parameter from the request safely.
+
+        :param request: The request containing the parameters.
+        :return: The value of the override_threshold parameter, or False if not present.
+        :rtype: bool
+        """
+        return request.params.get('override_threshold', False) if request.params else False
+
+    def get_recommendations(self, topic: Dict[str, Any],
+                            override_threshold=False) -> RecommendationsResponse:
+        """
+        Get recommendations for the given topic.
+
+        :param topic: A dictionary containing the topic for which to get recommendations. See
+        https://github.com/learningequality/le-utils/blob/main/spec/schema-embed_topics_request.json
+        :param override_threshold: A boolean flag to override the recommendation threshold.
+        :return: The recommendations for the given topic. :rtype: RecommendationsResponse
+        """
+
+        recommendations = []
+        request = EmbedTopicsRequest(
+            params={'override_threshold': override_threshold},
+            json=topic,
+        )
+
+        cached_response = self.response_exists(request)
+        if cached_response:
+            response = cached_response
+        else:
+            response = self.generate_embeddings(request=request)
+            if not response.error:
+                self.cache_embeddings_request(request, response)
+
+        nodes = self._extract_data(response)
+        if len(nodes) > 0:
+            node_ids = [node['contentnode_id'] for node in nodes]
+            recommendations = list(ContentNode.objects.filter(id__in=node_ids))
```

Comment:
We may not want to preemptively cast this to a list to get all the objects. Returning a queryset is very flexible. 

On the frontend, we'll need access to the data, either directly from the API or via the public API, as long as we have the data points required here: https://github.com/learningequality/studio/blob/unstable/contentcuration/contentcuration/frontend/channelEdit/vuex/contentNode/actions.js#L52C55-L52C82